assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -142,10 +146,39 @@ const PageEdit = ( {
 		}
 	}, [ mediaType, mediaUrl ] );
 
+	// If there is more than one immovable block, leave only the last and remove the others.
+	useEffect( () => {
+		if ( immovableBlocks.length > 1 ) {
+			immovableBlocks.pop();
+			const blocksToRemove = immovableBlocks.map( ( { clientId: blockId } ) => blockId );
+			removeBlocks( blocksToRemove );
+			const removeMessage = sprintf(
+				_n(
+					'%d block removed. Only one CTA/Attachment block allowed per Page.',
+					'%d blocks removed. Only one CTA/Attachment block allowed per Page.',
+					blocksToRemove.length,
+					'amp'
+				),
+				blocksToRemove.length
+			);
+			createErrorNotice(
+				removeMessage,
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+		}
+	}, [ childrenOrder, clientId, createErrorNotice, immovableBlocks, removeBlocks ] );
+
 	useEffect( () => {
 		if ( childrenOrder.length <= 1 ) {
 			return;
 		}
+		// If there is an illegal number of immovable blocks (more than 1).
+		if ( immovableBlocks.length > 1 ) {
+			return;
+		}
 		const ctaBlock = getCallToActionBlock( clientId );
```

Comment:
Couldn't all of this logic ([lines 178-193](https://github.com/ampproject/amp-wp/pull/3593/files#diff-3ebaa9cdbb7e5502ab37969c8f31350eR182-R191)) be reduced to:

```js
if (immovableBlocks.length === 1) {
  const blockToMove = immovableBlocks[0];
  // continue check for order and do move
}
```

Reply:
Good point! Should be. _Testing_

Reply:
We could actually use 
```
if ( immovableBlocks.length !== 1 ) {
			return;
		}
```
above, then we could just do `const blockToMove = immovableBlocks[0];`.

Reply:
Changed in 63850c9.